### PR TITLE
New tests to be run at deployment

### PR DESCRIPTION
**Pull request by chartjes**
Checks for expected fields in __version__ and __heartbeat__

### CODE REVIEW

File: config-test/test_server_details.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # Then make the we only have the expected fields in the data
+    for field in expected_fields:
+        assert field in data
+
+
+def test_heartbeat(conf, env):
+    response = requests.get(conf.get(env, 'server') + '/__heartbeat__')
+    data = response.json()
+
+    expected_fields = {'permission', 'cache', 'storage'}
```
Comment (Natim) [inline config-test/test_server_details.py]: In that case don't you need 'oauth'?

**Comment by chartjes** [comment]
The Kinto server that this test is run against that is on staging doesn't have the 'oauth' field in the heartbeat response:

https://kinto.stage.mozaws.net/v1/__heartbeat__

{"permission":true,"cache":true,"storage":true}

If it's supposed to have 'oauth' in production, then I guess I should store the expected fields in my manifest.ini file

**Comment by Natim** [comment]
Oh then we are not talking about kinto webextension anymore? I am a bit lost to be honest. How does one know which stack we are talking about?

**Comment by chartjes** [comment]
There will be separate repos for each Kinto "project". Kinto webextension has it's own set of tests that will be run whenever a deployment has been done. This set is for when kinto-dist (which I consider the main Kinto 'project') is deployed.

There will be some overlap but this is the approach QA has chosen as we build out our own testing infrastructure that will be triggered whenever a deployment of the various Kinto 'projects' happens.

**Comment by Natim** [comment]
Ok,
- https://github.com/Kinto/kinto-webextensions-tests for kinto-webextensions
- https://github.com/Kinto/kinto-integration-tests for kinto-settings

Is that correct?

**Comment by chartjes** [comment]
That is the idea

File: config-test/test_server_details.py (hunk 1 above)
Comment (Natim) [inline config-test/test_server_details.py]: How does one know that we are testing the reader here?

Comment (chartjes) [inline config-test/test_server_details.py]: Good point, I should make that clearer

File: config-test/test_server_details.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+@pytest.fixture
+def conf():
+    config = configparser.ConfigParser()
+    config.read('manifest.ini')
+    return config
+
+
+def test_version(conf, env):
+    response = requests.get(conf.get(env, 'reader_server') + '/__version__')
```
Comment (tarekziade) [inline config-test/test_server_details.py]: sounds like a good use case for smwogger

File: run [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-virtualenv venv
+virtualenv --python=`which python3` venv
```
Comment (Natim) [inline run]: Note that `virtualenv --python python3 venv` should also work

**Pull request merged**